include/KeyboardEvents.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <exception>
+#include <functional>
+#include <memory>
+#include <variant>
+
+#include <linux/input.h>
+
+#include "Device.h"
+#include "KeyTable.h"
+
+namespace hemiola
+{
+    /*!
+     * @brief struct which represents a keyboard event
+     */
+    struct KeyState
+    {
+        wchar_t key;
+        unsigned int repeats;  // count_repeats differs from the actual number of repeated
+        // characters! afaik, only the OS knows how these two values are
+        // related (by respecting configured repeat speed and delay)
+        bool repeatEnd;
+        input_event event;
+        bool scanCodeOk;
+        bool capslock = false;
+        bool shift = false;
+        bool altgr = false;
+        bool ctrl = false;  // used for identifying Ctrl+C / Ctrl+D
+    };
+
+    /*!
+     * @brief class for capturing keyboard events
+     */
+    class KeyboardEvents
+    {
+    public:
+        explicit KeyboardEvents ( std::shared_ptr<KeyTable> keyTable );
```

Comment:
Consider a simple wrapper class to aggregate the `KeyTable`, `KeyboardEvents`, and `Device` instances.  It seems like the lifespan of all of those objects are tightly coupled, and you could avoid the `shared_ptr` stuff.  In general I don't think the event handling code should have explicit shared ownership of those other objects.

Reply:
I separated things for easy testing. Also, this allows for future work using configs with specified device or key map.